### PR TITLE
Fixed #34565 -- Added support for async checking of user passwords.

### PR DESCRIPTION
**Pull request by HappyDingning**
Add `acheck_password` method and its unit tests. This method will call the async setter function to update the password field of the user table when the `settings.PASSWORD_HASHERS` is changed.

### CODE REVIEW

File: tests/async/test_async_checkpassword.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+class AsyncPasswordUpgradeTestCase(TestCase):
+    @override_settings(PASSWORD_HASHERS=PASSWORD_HASHERS)
+    async def test_check_password_upgrade(self):
+        """
+        async version of tests.auth_tests.test_models.AbstractUserTestCase.test_check_password_upgrade.
+        """
+        user = await sync_to_async(User.objects.create_user)(username="user", password="foo")
+        initial_password = user.password
+        self.assertTrue(await user.acheck_password("foo"))
```
Comment (felixxm) [inline tests/async/test_async_checkpassword.py:18]: Use `assertIs(..., True)` instead as described in [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).

**Review by felixxm — COMMENTED** [review]
@HappyDingning Thanks :+1: Please fix linter errors. Also, docs changes and release notes are missing. We should document `User.acheck_password()`, `AbstractBaseUser.acheck_password()`, and `acheck_password()`.

File: tests/async/test_async_checkpassword.py (hunk 1 above)
Comment (HappyDingning) [reply tests/async/test_async_checkpassword.py]: OK, no problem.

**Comment by HappyDingning** [comment]
@felixxm Hello, I implemented the `acheck_passport` document by referring to the document in the ORM section. I previewed the document after making changes and saw a green line appearing in front of some paragraphs. I want to ask if I need to try another way of writing?

**Comment by felixxm** [comment]
> @felixxm Hello, I implemented the `acheck_passport` document by referring to the document in the ORM section.

I'm not sure what `acheck_passport()` has to do with the ORM :thinking: Please submit doc updates and we will review.

> I previewed the document after making changes and saw a green line appearing in front of some paragraphs. I want to ask if I need to try another way of writing? (Sorry, I have some problems here, I will upload pictures as soon as possible.)

There is no need to upload pictures, just submit doc changes and we will review.

> In addition, regarding the release document, which version of the document should I write in? 4.2.2 or 5.0?

5.0

**Comment by HappyDingning** [comment]
> > @felixxm Hello, I implemented the `acheck_passport` document by referring to the document in the ORM section.
> 
> I'm not sure what `acheck_passport()` has to do with the ORM 🤔 Please submit doc updates and we will review.
> 
> > I previewed the document after making changes and saw a green line appearing in front of some paragraphs. I want to ask if I need to try another way of writing? (Sorry, I have some problems here, I will upload pictures as soon as possible.)
> 
> There is no need to upload pictures, just submit doc changes and we will review.
> 
> > In addition, regarding the release document, which version of the document should I write in? 4.2.2 or 5.0?
> 
> 5.0

Ok, thanks for your patient reply, I will upload the document as soon as possible

File: docs/topics/auth/passwords.txt [hunk 2]
```diff
@@ -478,6 +478,9 @@ to create and validate hashed passwords. You can use them independently
 from the ``User`` model.
 
 .. function:: check_password(password, encoded, setter=None, preferred="default")
+.. function:: acheck_password(password, encoded, setter=None, preferred="default")
+
+*Asynchronous version*: ``acheck_password()``
```
Comment (felixxm) [inline docs/topics/auth/passwords.txt:483]: `versionchanged` annotations are missing, e.g.
```
.. versionchanged:: 5.0

    ``acheck_password()`` method was added.
```

File: docs/releases/5.0.txt [hunk 3]
```diff
@@ -153,6 +153,16 @@ Minor features
...
 
+* ``AbstractBaseUser`` and ``User`` now add
+  :meth:`.AbstractBaseUser.acheck_password` and
+  :meth:`.User.acheck_password` to check a user's
+  password and refresh the data in the database asynchronously when
+  appropriate.
+
+* ``hashers`` now adds :meth:`django.contrib.auth.hashers.acheck_password` to
+  check a user's password and refresh the data in the database
+  asynchronously when appropriate.
```
Comment (felixxm) [inline docs/releases/5.0.txt:164]: Maybe:
```
* The new :func:`django.contrib.auth.hashers.acheck_password` function is an
  asynchronous version of ...
```
Also, it's a function not a method.

File: django/contrib/auth/hashers.py [hunk 4]
```diff
@@ -63,11 +61,32 @@ def check_password(password, encoded, setter=None, preferred="default"):
...
+    regenerate the password.
+    """
+    is_correct, must_update = verify_password(password, encoded, preferred=preferred)
+
     if setter and is_correct and must_update:
         setter(password)
     return is_correct
 
 
+async def acheck_password(password, encoded, asetter=None, preferred="default"):
```
Comment (felixxm) [inline django/contrib/auth/hashers.py:82]: Both functions should have the same signature
```suggestion
async def acheck_password(password, encoded, setter=None, preferred="default"):
```

**Comment by felixxm** [comment]
@HappyDingning Thanks :+1: Welcome aboard :boat: 

I pushed small edits.

**Comment by HappyDingning** [comment]
> @HappyDingning Thanks 👍 Welcome aboard ⛵
> 
> I pushed small edits.

Thanks a lot for helping me optimize my code.😊 I will check for checks failure issues as soon as possible

**Comment by felixxm** [comment]
> Thanks a lot for helping me optimize my code.blush I will check for checks failure issues as soon as possible

No need, this test is flaky,

**Comment by dingning-aspecta** [comment]
> > Thanks a lot for helping me optimize my code.blush I will check for checks failure issues as soon as possible
> 
> No need, this test is flaky,

Thank you very much for your review

**Pull request merged**